src/simplewallet/simplewallet.cpp

```diff
@@ -6856,7 +6850,12 @@ bool simple_wallet::sweep_main(uint64_t below, Transfer transfer_type, const std
   }
 
   cryptonote::address_parse_info info;
-  if (!cryptonote::get_account_address_from_str_or_url(info, m_wallet->nettype(), local_args[0], oa_prompter))
+  std::string default_addr = m_wallet->get_subaddress_as_str({m_current_subaddress_account, 0});
+  if (args_.size() > 0)
+  {
+    default_addr = local_args[0];
+  }
+  if (!cryptonote::get_account_address_from_str_or_url(info, m_wallet->nettype(), default_addr, oa_prompter))
```

Comment:
This is nice - and it matches up with the GUI wallet's ability to sweep all to myself without putting in an address (I don't use the GUI wallet often, but as I recall there's a button on the staking page that can do this).

A few comments:
- the test for whether a wallet was passed should be using `local_args` rather than `args_`.  (`local_args` has had all the other things that it understood pulled off such as priority, `index=`, `outputs=` arguments).  That way I can specify something like `sweep_all unimportant` to do a lowest priority sweep to myself.)
- `default_addr` is sort of a weird name because it starts out as the default address but then, if there's an argument, gets set to the (non-default) argument.
- There's no need to fetch the current wallet address with the `get_subaddress...()` call if there is a wallet address specified (since you're just going to replace it).  Better to only make the call only if you need it with something like:
```C++
std::string addr;
if (local_args.size() > 0)
  addr = local_args[0];
else
  addr = m_wallet->get_subaddr...;
```
- The usage and descriptions need an update to make the feature discoverable, i.e. putting `[` and `]` around the address and payment ID in the usage to show it as: `[<address> [<payment_id (obsolete)>]]` in the usage string, and then describe that it sweeps to your main address if omitted in the descriptions.  (Also note that this change affects both sweep_all and sweep_below, so both need an update!).